mkosi/distributions/debian.py

```diff
@@ -52,32 +50,55 @@ def repositories(state: MkosiState) -> list[str]:
 
     @classmethod
     def install(cls, state: MkosiState) -> None:
-        repos = {"main", *state.config.repositories}
+        # Instead of using debootstrap, we replicate its core functionality here. Because dpkg does not have
+        # an option to delay running pre-install maintainer scripts when it installs a package, it's
+        # impossible to use apt directly to bootstrap a Debian chroot since dpkg will try to run a maintainer
+        # script which depends on some basic tool to be available in the chroot from a deb which hasn't been
+        # unpacked yet, causing the script to fail. To avoid these issues, we have to extract all the
+        # essential debs first, and only then run the maintainer scripts for them.
+
+        # First, download the essential debs. Add usrmerge explicitly because it's not marked Essential on
+        # Ubuntu.
+        setup_apt(state, cls.repositories(state))
+        invoke_apt(state, "update")
+        invoke_apt(state, "install", ["--download-only", "?essential", "usrmerge"])
```

Comment:
Don't install usrmerge, that does a live conversion which is complex and has tons of dependencies. Instead, install usr-is-merged if it is available